### DISCUSSION TITLE
Setting unsigned integer oid value

### DISCUSSION THREAD

**Discussion by stsojithomas** [hh-h/aiosnmp#40] [Q&A] [answered]
Is there a way to set unsigned integer value to an oid?

Thanks!

**Comment by hh-h** [comment]
What happened if you pass unsigned integer via set? Could you add more details?

**Reply by stsojithomas** [reply]
`aiosnmp.exceptions.SnmpErrorWrongType: index: 1, oid: _oid_, message: The value specifies a type that is inconsistent with the type required for the variable.`

This is the error supplying int for an oid which is unsigned. 
Similar restrictions with using to set Gauge types (if they are modifiable)

**Reply by stsojithomas** [reply]
It has to do with tag 0x02 being passed for unsigned or gauge32 which the device rejects. Correct one is 0x42. 
I have a verified solution in my fork, which involves a custom data type for Gauge32, unless you have a different approach in mind. 
Please let me know your thoughts.

**Reply by stsojithomas** [reply]
In the changes, I changed decoder as well to Gauge32 to be consistent. However, that will break existing "reads" due to change from int to Gauge32 for those oids, unless caller updates their code to convert. 

Alternate is to keep decoder as existing, and change type for set/encoder only. Thoughts?

**Reply by hh-h** [reply]
Sorry for the long response, I'll take a look at this. Also, I updated pipeline and deps, so you can sync your PR with master if you want.

**Reply by stsojithomas** [reply]
Thanks! 
I have pulled the latest changes and the pipeline is running fine except for the pytests. I will fix the pytests once we decide on the decoder part (to leave it as int return type or match with encoder of Gauge32).

**Reply by hh-h** [reply]
I make initial PR, that's my vision, see https://github.com/hh-h/aiosnmp/pull/43
I'm not sure that it will be work without some encoders in rust, but I don't really have time to write them, so feel free to test my version and update it if needed.

**Reply by stsojithomas** [reply]
@hh-h Sorry for the delayed response. 
Validated int, gauge32, and string in various combinations, and they work. No changes needed to the encoder as when we pass the appropriate tag, the device accepts the value. 

The code may be merged and dist built.

Thank you!

**Comment by stsojithomas** [comment, marked as answer]
Fixed with release 0.7.2
